src/main/java/org/jabref/logic/cleanup/CleanupWorker.java

```diff
@@ -86,4 +103,17 @@ private CleanupJob toJob(CleanupPreferences.CleanupStep action) {
                     throw new UnsupportedOperationException(action.name());
         };
     }
+
+    private void showFileMoveExceptions(BibEntry entry) {
```

Comment:
This is too early.

Do following:

1. Add method `resetCleanUpFailures()` to `CleanupWorker`
2. Add field `List<JabRefExceptions> failures to CleanupWorker
3. Add method `getFailures()` to CleanupWorker
4. Modify `org.jabref.gui.cleanup.CleanupAction#cleanup` to first `resetCleanUpFailures`, then iterate through the entries and then show the errors

While you are in `resetCleanUpFailures`, move the `NamedCompund` generation and ending outside of the loop - the cleanup of entries is a batch and should not be split.